### DISCUSSION TITLE
Event to manipulate postEditUrl destination

### DISCUSSION THREAD

**Discussion by kringkaste** [craftcms/cms#11996] [Ideas] [answered]
If you have defined your own source, based on a filter of a field for example, then it would be great if you could be redirected back to the corresponding source after saving an entry (based of the value). Currently you always end up in the source "all entries".

To be more flexible an event in `postEditUrl` would be great that could override the current behavior.

**Comment by kringkaste** [comment]
As I just see, custom sources do not have their own URL, they always have the one of the section and therefore cannot be accessed directly at all. Maybe we could use the UID of the custom source?

**Comment by brandonkelly** [comment, marked as answer]
> As I just see, custom sources do not have their own URL, they always have the one of the section and therefore cannot be accessed directly at all. Maybe we could use the UID of the custom source?

Bingo. Just resolved this for Craft 4.3. Going forward element index pages will get a `source` query string param with the selected source key, and that will be referenced when determining which source to select by default. (a1a0eca2233024cade8a31c1a9d61e2091393313)

**Reply by kringkaste** [reply]
Great that we now can have deep links to sources. That's very helpful.

How about the second part of this idea? If I edit an entry in a (custom) source, it would be very nice to return to the previous selected source after saving the entry and not in "all entries". Our idea was to have an event to manipulate the postEditUrl() function of entries. So a plugin can decide on base of the entry, where the editor will be redirect to. :)

**Reply by brandonkelly** [reply]
Sorry, must have missed that. Agree that would be great. Will put it on the 4.4 list.

**Reply by brandonkelly** [reply]
Changed for 4.4! 379cdc51683c9ba22b17bc393b9018bcae06f5d8

**Reply by brandonkelly** [reply]
4.4 is out!
